### DISCUSSION TITLE
Running MuJoCo headless on servers

### DISCUSSION THREAD

**Discussion by pecey** [google-deepmind/mujoco#1064] [General]
Hi,

I was trying to run MuJoCo (using Gymnasium) in a headless mode on our server with CPUs, but whenever the code tries to `import mujoco`, it fails with the following error: 
```
File "/home/u000/py3.9.8/lib/python3.9/site-packages/mujoco/egl/__init__.py", line 67, in <module>
    raise ImportError(
ImportError: Cannot initialize a EGL device display. This likely means that your EGL driver does not support the PLATFORM_DEVICE extension, which is required for creating a headless rendering context.
```
I have ~~`MUJOCO_EGL`~~ `MUJOCO_GL` and `PYOPENGL_PLATFORM` set. I am not sure what am I missing. Any pointers to what I should be looking at or workarounds?

**Comment by kevinzakka** [comment]
The variable is `MUJOCO_GL`.

**Reply by pecey** [reply]
Sorry, that was a typo and should have been MUJOCO_GL. Fixed it. I have `MUJOCO_GL=egl` set.

**Reply by kevinzakka** [reply]
Do you have a GPU?

**Reply by saran-t** [reply]
What GPU do you have?

**Reply by pecey** [reply]
No I was trying to run on a CPU-based server. Does MuJoCo require GPU to run? I believe earlier it used to run on CPU-based systems as well.

**Comment by pecey** [comment]
This is probably an issue with drivers at the compute node. It was working on the login node of the cluster but not on the compute nodes. Switching to `osmesa` worked for me.
